### ISSUE TITLE
Using value-pattern with the color from typography field

### ISSUE THREAD

**Issue by joakimaglo** [themeum/kirki#2169]
### Issue description:
I would like to use the color from a Typography field to set a color within a value_pattern.

**This works within a color field:**
```
array(
  'element' => array(
    '.site a',
  ),
  'property' => 'box-shadow',
  'value_pattern' => 'inset 0 -2px 0 $',
)
```
**...but I can't get this to work within a typography field:**
```
array(
  'element' => array(
    '.site a',
  ),
  'property' => 'box-shadow',
  'choice'   => 'color',
  'value_pattern' => 'inset 0 -2px 0 $',
)
```
Am I doing something wrong or is this not supported?

Thanks!

### Version used:
3.0.40

### Using theme_mods or options?
theme_mods

### Code to reproduce the issue (config + field(s))
See above.

**Comment by DannyCooper** [comment]
Confirmed bug.

This is strange, the output looks something like this:

![image](https://user-images.githubusercontent.com/152804/59963200-a1895900-94e7-11e9-86d8-e09f64705541.png)

When it should be:

![image](https://user-images.githubusercontent.com/152804/59963209-b5cd5600-94e7-11e9-9c34-4a2b13af4cdb.png)

Code: 

```
Kirki::add_field( 'prefix_config', [
	'type'        => 'typography',
	'settings'    => 'my_typography_setting',
	'label'       => esc_html__( 'Control Label', 'kirki' ),
	'section'     => 'title_tagline',
	'default'     => [
		'font-family'    => 'Roboto',
		'variant'        => 'regular',
		'color'          => '#333333',
	],
	'priority'    => 10,
	'transport'   => 'auto',
	'output' => array(
		array(
			'choice'   => 'color',
			'element'  => '.menu-item a',
			'property' => 'box-shadow',
			'value_pattern' => 'inset 0 -2px 0 $',
		),
	)
] );
```

**Comment by aristath** [comment]
Fixed, will be included in v4.0 :+1:
